### PR TITLE
Add Dataspace Task

### PR DESCRIPTION
**Pull request by CBists**
Command change:
CREATE DATASPACE ds1 - create new dataspace with id = ds1
ADD SHARDING RULE r1 DATASPACE ds1 TABLE test1 COLUMNS age - add new sharding rule with link to ds1
ADD KEY RANGE krid11 FROM 1 TO 18 ROUTE TO sh1 DATASPACE ds1; - add new key range with link to ds1

old version commands(without dataspace) saved

to test, we can use this example:

ds1:
test1.age:
1-18 -> sh1
19-100 -> sh2
101-3000 -> sh3

ds2:
test1.age:
1-1000 -> sh1
1001 - 2000 -> sh2
2001 - 3000 -> sh3

*connect to test1 table, ds1 dataspace, db1 database*
INSERT INTO test1(age) VALUES(10)
INSERT INTO test1(age) VALUES(50)
INSERT INTO test1(age) VALUES(500)
INSERT INTO test1(age) VALUES(1500)
INSERT INTO test1(age) VALUES(2500)

*connect to test1 table, ds2 dataspace, db1 database*
INSERT INTO test1(age) VALUES(10)
INSERT INTO test1(age) VALUES(50)
INSERT INTO test1(age) VALUES(510)
INSERT INTO test1(age) VALUES(1500)
INSERT INTO test1(age) VALUES(2500)

in db should:
sh1:             sh2:                sh3:
10                50                  500
10                1500              1500
50                                      2500
500                                    2500

### CODE REVIEW

File: yacc/console/gram.y [hunk 1]
```diff
@@ -346,6 +347,11 @@ drop_stmt:
     }
 
 add_stmt:
+    ADD dataspace_define_stmt
```
Comment (reshke) [inline yacc/console/gram.y:350]: indentation
we have no linters here

File: yacc/console/gram.y [hunk 2]
```diff
@@ -480,20 +486,25 @@ sharding_rule_hash_function_clause:
 	}
 	| /*EMPTY*/ { $$ = ""; }
 
+link_dataspace:
```
Comment (reshke) [inline yacc/console/gram.y:489]: opt_dataspace

File: yacc/console/gram.y [hunk 3]
```diff
@@ -421,18 +427,18 @@ dataspace_define_stmt:
 	}
 
 sharding_rule_define_stmt:
-	SHARDING RULE any_id sharding_rule_table_clause sharding_rule_argument_list
+	SHARDING RULE any_id link_dataspace sharding_rule_table_clause sharding_rule_argument_list
```
Comment (reshke) [inline yacc/console/gram.y:430]: idk maybe
`CREATE SHARDING RULE r1 DATASPACE ds1` is not sqlish enough. 
IMO `CREATE SHARDING RULE r1 USING DATASPACE ds1` or `CREATE SHARDING RULE r1 FOR DATASPACE ds1` would be better

File: protos/sharding_rules.proto [hunk 4]
```diff
@@ -22,7 +22,9 @@ message AddShardingRuleRequest {
 
 message AddShardingRuleReply {}
 
-message ListShardingRuleRequest {}
+message ListShardingRuleRequest {
```
Comment (reshke) [inline protos/sharding_rules.proto:25]: Do we have implementation for this? 
Also, we should support `LIST DATASPACE` in spqr-console

File: yacc/console/gram.y (hunk 2 above)
Comment (CBists) [reply yacc/console/gram.y:489]: Changed

File: protos/sharding_rules.proto (hunk 4 above)
Comment (CBists) [reply protos/sharding_rules.proto:25]: Added

File: yacc/console/gram.y (hunk 1 above)
Comment (CBists) [reply yacc/console/gram.y]: fixed

File: yacc/console/gram.y (hunk 3 above)
Comment (CBists) [reply yacc/console/gram.y]: Changed. Now declaring:
CREATE SHARDING RULE r1 FOR DATASPACE ds1

**Comment by CBists** [comment]
Right now we can switch dataspace by using command SET DATASPACE *dataspaceId* and we can print all existing dataspaces by using command SHOW DATASPACES in spqr-console

**Pull request closed without merging**